app/lib/three_scale/search.rb

```diff
@@ -98,14 +100,11 @@ def order_by(column = nil, direction = nil)
         end
       end
 
-      # scope_search is a bad name name, but #search is already taken by thinking-sphinx
-      def scope_search(params, reduce = true)
-        return default_search_scope unless allowed_scopes || default_search_scopes.present?
-        return default_search_scope unless params
+      def sphinx_search(params)
+        return default_search_scope unless allowed_scopes || default_search_scopes.present? || params
 
         params = params.dup
-        # this escapes sphinx special characters like $
-        params[:query] = Riddle.escape(params[:query]) if params[:query]
+        params[:query] = escape_query(params)
```

Comment:
Original version is more readable to me and intention is clear.

Reply:
OK